keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -789,79 +732,54 @@ class ClusterHandler(
     atZone(ZoneId.systemDefault()).format(DateTimeFormatter.ISO_DATE_TIME)
 
   /**
-   * Translates a LaunchConfiguration object to a ClusterSpec.LaunchConfigurationSpec with default values omitted for export.
+   * Translates a ServerGroup object to a ClusterSpec.ServerGroupSpec with default values omitted for export.
    */
-  private fun LaunchConfiguration.exportSpec(account: String, application: String): ClusterSpec.LaunchConfigurationSpec {
-    val defaultKeyPair = cloudDriverCache.defaultKeyPairForAccount(account)
-    return ClusterSpec.LaunchConfigurationSpec(
-      instanceType = instanceType,
-      ebsOptimized = if (!ebsOptimized) {
-        null
-      } else {
-        ebsOptimized
-      },
-      // for the iam role, we compare against a default based on a stable naming convention
-      iamRole = if (iamRole == LaunchConfiguration.defaultIamRoleFor(application)) {
-        null
-      } else {
-        iamRole
-      },
-      // for the key pair, we compare against the currently configured default in clouddriver since there are
-      // multiple naming conventions, and handle the case where the default includes a region placeholder
-      keyPair = if (defaultKeyPair.contains(REGION_PLACEHOLDER)) {
-        if (defaultKeyPair.replace(REGION_PLACEHOLDER, REGION_PATTERN).toRegex().matches(keyPair)) {
-          null
-        } else {
-          keyPair
-        }
-      } else {
-        if (keyPair == defaultKeyPair) {
-          null
-        } else {
-          keyPair
-        }
-      },
-      instanceMonitoring = if (!instanceMonitoring) {
+  private fun ServerGroup.exportSpec(account: String, application: String): ClusterSpec.ServerGroupSpec {
+    val defaults = ClusterSpec.ServerGroupSpec(
+      capacity = Capacity(1, 1, 1),
+      dependencies = ClusterDependencies(),
+      health = Health().exportSpec(),
+      scaling = Scaling(),
+      tags = emptyMap()
+    )
+
+    val thisSpec = ClusterSpec.ServerGroupSpec(
+      launchConfiguration = launchConfiguration.exportSpec(account, application),
+      capacity = capacity,
+      dependencies = dependencies,
+      health = health.exportSpec(),
+      scaling = if (!scaling.hasScalingPolicies()) {
         null
       } else {
-        instanceMonitoring
+        scaling
       },
-      ramdiskId = ramdiskId
+      tags = tags
     )
+
+    // it's safe to assume a non-null result here because not all properties have defaults
+    return buildSpecFromDiff(defaults, thisSpec)!!
```

Comment:
I dislike using `!!`. Can we instead throw an exception if it's null? like `return buildSpecFromDiff(defaults, thisSpec) ?: throw OhNoException()`

Reply:
or `return checkNotNull(buildSpecFromDiff(defaults, thisSpec))`